shared/src/commonMain/kotlin/org/itmodreamteam/myrest/shared/restaurant/RestaurantClientImpl.kt

```diff
@@ -12,6 +12,13 @@ class RestaurantClientImpl : RestaurantClient {
 
     private val client = HttpClientProvider.provide()
 
+    override suspend fun getById(id: Long): RestaurantInfo {
+        return client.get {
+            url("/restaurants/$id")
+            provideAccessToken()
```

Comment:
This is public method, authentication mustn't be required. Otherwise, if we were browsing restaurants unauthenticated and then tried to open a restaurant page, we'd be required to sign in.